### ISSUE TITLE
[jvm-packages] For training data with group, empty RDD partition threw exception

### ISSUE THREAD

**Issue by weitian** [dmlc/xgboost#3749]
when using spark training on training data with group:
XGBoost.trainDistributed(queriesRDD, params, round, nWorkers, obj, eval, useExternalMemory, missing, hasGroup=true)

The spark job failed with exception:
18/10/03 11:02:04 ERROR executor.Executor: Exception in task 117.0 in stage 2.0 (TID 532)
java.lang.ArrayIndexOutOfBoundsException: 0
	at ml.dmlc.xgboost4j.scala.spark.LabeledPointGroupIterator.next(XGBoost.scala:527)
	at ml.dmlc.xgboost4j.scala.spark.LabeledPointGroupIterator.next(XGBoost.scala:493)
	at scala.collection.Iterator$$anon$13.hasNext(Iterator.scala:462)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at org.apache.spark.util.collection.ExternalSorter.insertAll(ExternalSorter.scala:199)
	at org.apache.spark.shuffle.sort.SortShuffleWriter.write(SortShuffleWriter.scala:63)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:96)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:53)
	at org.apache.spark.scheduler.Task.run(Task.scala:108)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:338)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

The cause is that my training data does have empty partitions.

**Comment by weitian** [comment]
It is a quick fix, will submit a PR later

**Comment by hcho3** [comment]
@weitian Thanks. You should add yourself to CONTRIBUTORS.md when you do so.

**Comment by hcho3** [comment]
Closed by #3750
